### PR TITLE
Synth job addition

### PR DESCRIPTION
**Pull request by SpaceLoveSs13**
## About The Pull Request

Adds Synthetic job slot every 40 pop.

## Why It's Good For The Game

As we are constantly moving towards high pop, I believe synthetics getting more slot every 40 pop makes sense as generally there is a lack of command. Synthetics can do most shipside job, and as the pop increases, it generally gets difficult to manage shipside. Addition of a support role of Synthetic will ensure this.

I may change the pop number upon testing and feedback. This needs testing.

## Changelog
:cl: SpaceLove
balance: Gives 1 Synth job slot every 40 pop.

/:cl:

### CODE REVIEW

File: code/datums/gamemodes/infestation.dm [hunk 1]
```diff
@@ -6,3 +6,10 @@
 		return
 	var/datum/job/scaled_job = SSjob.GetJobType(/datum/job/terragov/squad/smartgunner)
 	scaled_job.job_points_needed  = 20 //For every 10 marine late joins, 1 extra SG
+
+/datum/game_mode/infestation/scale_roles()
```
Comment (BraveMole) [inline code/datums/gamemodes/infestation.dm:10]: you are duplicating a proc that exist already. Add code the proc that's just above

File: code/datums/gamemodes/infestation.dm [hunk 2]
```diff
@@ -6,3 +6,10 @@
...
+	. = ..()
+	if(!.)
+		return
+	var/datum/job/scaled_job = SSjob.GetJobType(/datum/job/terragov/synth)
+	scaled_job.job_points_needed  = 40 //For every 40 marines there will be 1 extra Synth.
```
Comment (BraveMole) [inline code/datums/gamemodes/infestation.dm:15]: This seems to be a bad way of doing that. Go look at how xeno job are done, also SG. 

it uses jobworth.

File: code/datums/jobs/access.dm [hunk 3]
```diff
@@ -343,5 +343,11 @@
 			. = size ? "MERC " : "MERC Engineer"
 		if("VM")
 			. = size ? "VAT " : "VatGrown Marine"
+		if("CS")
+			. = size ? "CS " : "Chief Synthetic"
+		if("MS")
+			. = size ? "MS" : "Major Synthetic"
+		if("SO1")
+			. = size ? "SO1 " : "Synthetic First Class"
```
Comment (BraveMole) [inline code/datums/jobs/access.dm:351]: synthetic having military ranks is hella weird. I prefere the MK thing

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/datums/jobs/access.dm (hunk 3 above)
Comment (SpaceLoveSs13) [reply code/datums/jobs/access.dm:351]: Yea old non updated branch.

**Pull request closed without merging**